examples/apps/heart_disease_eval/client/heart_gui.py

```diff
@@ -0,0 +1,493 @@
+
+		self.button_list = []
+		for i in range(len(options)):
+			button = tk.Radiobutton(master, text=options[i][0], 
+				variable=self.var, value=options[i][1])
+			self.button_list.append(button)
+			if i==0:
+				button.grid(row=cur_row, column=1, pady=(5,0), sticky="w")
+			else:
+				button.grid(row=cur_row, column=1, sticky="w")
+			cur_row += 1
+
+	def get(self):
+		try:
+			return self.var.get()
+		except tk.TclError:
+			return None
+
+	def enable(self):
+		for button in self.button_list:
+			button.config(state=tk.NORMAL)
+
+	def disable(self):
+		for button in self.button_list:
+			button.config(state=tk.DISABLED)
+
+class resultWindow(tk.Toplevel):
+	def __init__(self, parent, message):
+		tk.Toplevel.__init__(self, parent)
+		self.config(background="pale goldenrod")
+		self.parent = parent
+		# Lock main window
+		self.transient(parent)
+		self.grab_set()
+		self.initial_focus = self
+		self.initial_focus.focus_set()
+		self.title("Evaluation Result")
+		self.protocol("WM_DELETE_WINDOW", self.close)
+
+		# Main content
+		self.main_frame = tk.Frame(self, background="pale goldenrod")
+		self.main_frame.pack()
+
+		self.frame1 = tk.Frame(self.main_frame)
+		self.frame1.pack(side=tk.LEFT)
+		self.result_text = tk.StringVar()
+		self.label = tk.Label(self.frame1, textvariable=self.result_text,
+			width=45, background="pale goldenrod")
+		default_font = font.Font(font="TkDefaultFont")
+		new_font = default_font
+		new_font.config(weight=font.BOLD)
+		self.label.config(font=new_font)
+		self.label.pack()
+		self.frame2 = tk.Frame(self.main_frame, background="pale goldenrod")
+		self.frame2.pack(side=tk.LEFT)
+
+		# JSON sidebar
+		self.request_button = tk.Button(
+			self.frame2, text="View Request", command=self.request)
+		self.request_button.pack(fill=tk.X, padx=(0,10), pady=(10,0))
+
+		self.result_button = tk.Button(
+			self.frame2, text="View Result", command=self.result)
+		self.result_button.pack(fill=tk.X, padx=(0,10),pady=(10,0))
+
+		self.receipt_button = tk.Button(
+			self.frame2, text="View Receipt", command=self.receipt)
+		self.receipt_button.pack(fill=tk.X, padx=(0,10),pady=(10,0))
+
+		# Close button
+		self.close_button = tk.Button(self, text="Close", command=self.close)
+		self.close_button.pack(pady=(0,5))
+
+		self.evaluate(message)
+
+	# Submit work order and retrieve result
+	def evaluate(self, message):
+		self.result_text.set("Waiting for evaluation result...")
+		self.update()
+
+		# Create, sign, and submit workorder 
+		# Convert workloadId to hex
+		workload_id = "heart-disease-eval"
+		workload_id = workload_id.encode("UTF-8").hex()
+		wo_submit_json = jrpc_request.WorkOrderSubmitJson(3, 6000, "pformat",
+			worker_id, workload_id, "0x1234", 
+			worker_encryption_key=base64.b64decode(
+				worker_obj.worker_encryption_key).hex(),
+			data_encryption_algorithm="AES-GCM-256")
+		wo_id = wo_submit_json.get_work_order_id()
+		data_hash = "dataHash".encode("UTF-8").hex()
```

Comment:
Data hash should be actual hash value of data. Looks like existing sample code  is not correctly calculating it. You could keep it empty instead of using dummy value as its optional as per spec or calculate real hash value.

Reply:
Done.